pymc4/mcmc/samplers.py

```diff
@@ -822,3 +827,28 @@ def vectorized_logpfn(*state):
 
 def tile_init(init, num_repeats):
     return [tf.tile(tf.expand_dims(tens, 0), [num_repeats] + [1] * tens.ndim) for tens in init]
+
+
+def calculate_log_likelihood(
+    model: Model, posterior: Dict[str, tf.Tensor], sampling_state: flow.SamplingState
+) -> Dict[str, tf.Tensor]:
+    """Compute log likelihood by vectorizing chains and draws."""
+
+    def extract_log_likelihood(values, observed_rv):
+        st = flow.SamplingState.from_values(values, observed_values=sampling_state.observed_values)
+        _, st = flow.evaluate_model_transformed(model, state=st)
+        try:
+            dist = st.continuous_distributions[observed_rv]
+        except KeyError:
+            dist = st.discrete_distributions[observed_rv]
+        return dist.log_prob(dist.model_info["observed"])
+
```

Comment:
I am not sure here about having deterministic distributions as observed variables. If that's the case, we do need to handle that as well.